### PR TITLE
Add application considerations.

### LINKED ISSUE #173: Add application consideration section
We currently have _a lot_ of knobs for applications and wrapper protocols choosing to implement OPAQUE. We should add text which discusses the different knobs, how they might be controlled, and how higher-level APIs might be presented to applications.

### PR DESCRIPTION
**Pull request by chris-wood**
Closes #173 and #204.

### CODE REVIEW

File: draft-irtf-cfrg-opaque.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ author:
     name: Hugo Krawczyk
     organization: Algorand Foundation
     email: hugokraw@gmail.com
+ -
+    ins: D. Bourdrez
+    name: Daniel Bourdrez
+    email: d@bytema.re
```
Comment (bytemare) [inline draft-irtf-cfrg-opaque.md:23]: ❤️

**Comment by bytemare** [comment]
Do you mean with regard to #173?

**Comment by chris-wood** [comment]
@bytemare I also included #204 changes here. (Do you think more needs to be said to close out that issue?)

File: draft-irtf-cfrg-opaque.md [hunk 2]
```diff
@@ -1820,9 +1852,18 @@ Applications must use the same envelope mode when using this prevention
...
-of a simulated credential response message for an unregistered client if these client
-enumeration attacks can be mitigated through other application-specific means.
+Note that server implementations may choose to forego the construction of a
+simulated credential response message for an unregistered client if these client
+enumeration attacks can be mitigated through other application-specific means
+or are otherwise not applicable for their threat model.
+
+Finally, OPAQUE does not prevent prevent enumeration via the registration flow.
+For example, if an attacker tries to register with a given client identity that
+is also in use, the server will not complete the protocol. This allows an attacker
```
Comment (bytemare) [inline draft-irtf-cfrg-opaque.md:1862]:
```suggestion
is already in use, the server will not complete the protocol. This allows an attacker
```

**Review by bytemare — CHANGES_REQUESTED** [review]

**Review by bytemare — APPROVED** [review]

File: draft-irtf-cfrg-opaque.md [hunk 3]
```diff
@@ -1820,9 +1852,18 @@ Applications must use the same envelope mode when using this prevention
...
+to use the server's response during registration as an oracle for whether a
+given client identity is registered. Applications should mitigate against
```
[the remark this answers is not in the document]
Comment (chris-wood) [reply draft-irtf-cfrg-opaque.md]: I think this is resolved?

[the remark this answers is not in the document]
Comment (bytemare) [reply draft-irtf-cfrg-opaque.md]: Yes it is -- I resolved this yesterday when we talked about it

**Review by kevinlewi — APPROVED** [review]
This is very well-written. Thank you!

**Pull request merged**